### ISSUE TITLE
Geometry_Engine: Add 'align planes' or similarly named method to align plane rotations

### ISSUE THREAD

**Issue by rwemay** [BHoM/BHoM_Engine#1757]
#### Description:

Allow input of a vector to align the x-axis of a plane by rotating it about the plane normal/origin. Would allow us to (for example) align the planes created in the screenshot below. Whether or not this is also an issue for the plane constructor that takes 3 points is a separate issue. @pawelbaran , @al-fisher thoughts on this? 

![image](https://user-images.githubusercontent.com/15233608/81596184-881d2400-93bb-11ea-9482-a9bf4f5bbc9b.png)

**Comment by pawelbaran** [comment]
Yeah I like the idea (there has already been a similar thread I think). However, this would need to be thought through as this can easily get tricky (which local axis should be aligned, which is fixed, what if the axis cannot be aligned by rotating around one of the local axes etc.).

**Comment by rwemay** [comment]
The implementation I'm thinking of is specifically to align the X-axis of a plane to a guide vector by rotating it around it's normal/'z-axis'.

**Comment by pawelbaran** [comment]
Yeah, but would be nice to allow aligning other axes around others too, plus we need to make sure that the solution exists etc. Having said that, I am not against at all, just trying to think of what needs to be done :p

BTW, following @al-fisher just speaking on the morning stand-up, BHoM plane does not have X axis 🤣 So you would need to switch to `CoordinateSystem.Cartesian` I am afraid.

**Comment by al-fisher** [comment]
Thanks @pawelbaran 
Yes, sorry @rwemay - in BHoM our `CoordinateSystem.Cartesian` is the equivalent of the Grasshopper "Plane". 
We have explicitly made the distinction between a `Geometry.CoordinateSystem.Cartesian`  and the `Geometry.Plane` - the latter of which is defined only by origin and normal.

_(We also of course have a `Geometry.Basis` as well which has no origin or location.)_

So in BHoM, this issue should be align Coordinate Systems

**Comment by rwemay** [comment]
Right yep, sorry, had forgotten we'd gone with the literal definition - I did mean coordinate system - will edit issue title for clarity.
